src/main/kotlin/uk/gov/justice/digital/hmpps/educationemploymentapi/resource/ProfileResourceController.kt

```diff
@@ -96,11 +110,25 @@ class ProfileResource(
       )
     ]
   )
-  suspend fun createOffenderProfile(
+  fun createOffenderProfile(
     @PathVariable offenderId: String,
     @RequestBody requestDTO: ReadinessProfileRequestDTO,
     @AuthenticationPrincipal oauth2User: String
-  ): ReadinessProfileDTO = ReadinessProfileDTO(profileService.createProfileForOffender(oauth2User, offenderId, requestDTO.bookingId, requestDTO.profileData))
+  ): ReadinessProfileDTO {
+
+    validateOffenderId(offenderId)
```

Comment:
The auto wiring of the validator, and its direct invocation were lifted from the example branch that used R2DBC.
They were only there because R2DBC precluded the use of the Spring @validation-annotations.
The reason for moving to JPA was so that the direct invocation would not be needed, and yet it still all there?
I suggest removing these now and annotating the path params args, and allow Spring to auto validate as it should.